metrics-datadog/src/main/java/org/coursera/metrics/datadog/DatadogReporter.java

```diff
@@ -73,6 +76,15 @@ public void report(SortedMap<String, Gauge> gauges,
                      SortedMap<String, Timer> timers) {
     final long timestamp = clock.getTime() / 1000;
 
+    if (tagsCallback != null) {
+      List<String> dynamicTags = tagsCallback.getTags();
+      if (dynamicTags != null && ! dynamicTags.isEmpty()) {
+        List<String> newTags = TagsMerger.mergeTags(tags, tagsCallback.getTags());
+        tags.clear();
+        tags.addAll(newTags);
```

Comment:
I agree. That was my first thought but when I saw all those reportXXX methods are referencing tags directly, I switched to this route. Luckily, all those reportXXX are private and I'll change the api to include tags as parameter.